test/functional/sqlsrv/TC32_DeleteQuery.phpt

```diff
@@ -6,47 +6,49 @@ validates the outcome reported by "sqlsrv_rows_affected".
 --ENV--
 PHPT_EXEC=true
 --SKIPIF--
-<?php require('skipif.inc'); ?>
+<?php require('skipif_versions_old.inc'); ?>
 --FILE--
 <?php
 require_once('MsCommon.inc');
 
-function DeleteQuery()
+function deleteQuery()
 {
     $testName = "Statement - Delete Query";
     startTest($testName);
 
     setup();
-    $conn1 = connect();
+    $conn1 = AE\connect();
 
     $noRows = 10;
     $tableName = 'TC32test';
-    createTable($conn1, $tableName);
-    $noRowsInserted = insertRows($conn1, $tableName, $noRows);
+    AE\createTestTable($conn1, $tableName);
+    // Insert some random rows
+    $noRowsInserted = AE\insertTestRows($conn1, $tableName, $noRows);
 
     $row = 1;
     $keyValue = "0";
     while ($row <= $noRowsInserted) {
-        $stmt1 = selectFromTable($conn1, $tableName);
+        $stmt1 = AE\selectFromTable($conn1, $tableName);
         if (sqlsrv_fetch($stmt1) === false) {
-            fatalError("Failed to retrieve 1st row of data from test table");
+            fatalError("Failed to retrieve 1st row of data from test table", true);
         }
         $keyValue = sqlsrv_get_field($stmt1, 0, SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_CHAR));
         sqlsrv_free_stmt($stmt1);
 
         trace("Deleting rows from $tableName ...");
         $delRows = 1;
         if (strlen($keyValue) == 0) {
-            $stmt2 = executeQuery($conn1, "DELETE TOP(1) FROM [$tableName]");
+            $stmt2 = AE\executeQuery($conn1, "DELETE TOP(1) FROM [$tableName]");
             $cond = "(top row)";
         } else {
-            $cond = "(c1_int = $keyValue)";
-
-            $stmt3 = selectFromTableEx($conn1, $tableName, $cond);
+            $cond = "(c1_int = ?)";
```

Comment:
Is it a good idea to change the test to use a placeholder completely? Now even if isColEnc() is false, the test still uses sqlsrv_prepare with placeholder to run the query, whereas before is used to use sqlsrv_query on a direct query. When refactoring tests, we want to keep the behavior of the original test as intact as possible.

Reply:
I decided to put the logic in AE\executeQuery() instead so that each placeholder will be replaced with literal value if not encrypted